internal/controllers/reconciliation/controller.go

```diff
@@ -156,19 +157,20 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 	}
 
 	// Store the results
+	if modified {
```

Comment:
We now re-enter the loop _before_ writing status instead of directly after. This ensures that we don't write status that differs from what we get back on the next pass, so less risk of oscillation.